synapse/storage/databases/main/transactions.py

```diff
@@ -134,16 +134,17 @@ async def set_received_txn_response(
             response_dict: The response, to be encoded into JSON.
         """
 
-        await self.db_pool.simple_insert(
+        await self.db_pool.simple_upsert(
             table="received_transactions",
-            values={
+            keyvalues={
                 "transaction_id": transaction_id,
                 "origin": origin,
+            },
+            values={
```

Comment:
Should this be `values` or `insertion_values`? It seems like `insertion_values` would match the previous behavior better?

Reply:
```suggestion
            values={},
            insertion_values={
```